docs/simpleinjector/metricsCapturing/applyingMetricsCapturing.md

```diff
@@ -0,0 +1,65 @@
+
+The metrics-capturing decorator works by wrapping invocations of handler implementations with the following method calls:
+- `OnInvocationStart` (before invoking the handler)
+- `OnInvocationCompletedSuccessfully` (after invoking the handler and having it complete successfully)
+- `OnInvocationException` (after invoking the handler and having it fail due to an unhandled exception) 
+
+Here is an example query parameter object and query handler:
+
+```
+// a Functional query parameter object
+// assume SystemSettingsForCompany is a POCO (plain old C# object)
+// query returns Option<SystemSettingsForCompany> because it is possible that a company entity ID is invalid and no result would be returned in that case
+// constructor boilerplate has been left out for brevity
+public class GetSystemSettingsForCompanyQuery : IQuery<Result<Option<SystemSettingsForCompany>, Exception>>
+{
+    public int CompanyEntityID { get; }
+}
+```
+
+```
+// a Functional query handler implementation
+// actual implementation has not been included as it is not relevant to caching 
+public class GetSystemSettingsForCompanyQueryHandler : IQueryHandler<GetSystemSettingsForCompanyQuery, Result<Option<SystemSettingsForCompany>, Exception>>
+{
+    public Result<Option<SystemSettingsForCompany>, Exception> Handle(GetSystemSettingsForCompanyQuery query) => ...;
+}
+```
+
+Note that this `IQueryHandler<TQuery, TResult>` implementation defines `TQuery` as `GetSystemSettingsForCompanyQuery` and `TResult` as `Result<Option<SystemSettingsForCompany>, Exception>`.  To apply metrics-capturing to the query handler implementation shown above, we define a `IMetricsCapturingStrategyForQuery<TQuery, TResult>` implementation with `TQuery` and `TResult` types that are identical to those defined for the query handler implementation, like so:
+
+```
+// the Functional.CQS.AOP.MetricsCapturing.IMetricsCapturingStrategyForQuery implementation
+// it is required for metrics-capturing to be applied to the query handler implementation above
+public class GetSystemSettingsForCompanyMetricsCapturingStrategy : IMetricsCapturingStrategyForQuery<GetSystemSettingsForCompanyQuery, Result<Option<SystemSettingsForCompany>, Exception>>
+{
+    // use constructor injection to inject component dependencies required for metrics-capturing
+    
+    public void OnInvocationStart(GetSystemSettingsForCompanyQuery)
+    {
+        // perhaps post counts to Hosted Graphite so that 'number of calls' can be analyzed  
+    }
+
+    public void OnInvocationCompletedSuccessfully(GetSystemSettingsForCompanyQuery parameters, Option<SystemSettingsForCompany> result, TimeSpan timeElapsed)
+    {
+        // perhaps post performance timings to Hosted Graphite
+    }
+
+    public void OnInvocationException(TQuery parameters, Exception exception, TimeSpan timeElapsed)
```

Comment:
`GetSystemSettingsForCompanyQuery` instead of `TQuery`?

Reply:
Yep!